metadata-db-app/src/cmr/metadata_db/data/oracle/concepts.clj

```diff
@@ -224,6 +224,80 @@
        :existing-concept-id concept_id
        :existing-native-id native_id})))
 
+(defn validate-collection-not-associated-with-another-variable-with-same-name
+  "Validates that collection in the concept is not associated with a different
+  variable, which has the same name as the variable in the concept.
+  Returns nil if valid and an error response if invalid."
+  [db concept]
+  (let [variable-concept-id (get-in concept [:extra-fields :variable-concept-id])
+        associated-concept-id (get-in concept [:extra-fields :associated-concept-id])]
+    (when (and variable-concept-id associated-concept-id))
+      ;; Get all the other variables sharing the same name; not deleted
+      ;; and are associated with the same collection.
+      (let [stmt [(format "select va.variable_concept_id
+                           from   cmr_variable_associations va, 
+                                  (select concept_id, max(revision_id) maxrev
+                                   from cmr_variable_associations group by concept_id) latestva,
+                                  cmr_variables v
+                           where  va.revision_id = latestva.maxrev
```

Comment:
same logic error here with the variable association revisions. Also I think you could get the variable_name from the concept without querying the cmr_variables table. Also this query will not work for the case when there is another variable with the same name in a different provider (BTW, you should have a test covering this case).

Reply:
I'm not sure an association concept will contain the variable_name. I don't  think it does because the cmr_variable_associations table doesn't contain the variable_name. even if it does, I still need to query the cmr_variables table to get "other" variables with the same name.
Regarding the provider, I checked the provider constraint before checking the name constraint, so it's guaranteed that the variable and the collection are from the same provider. and the existing associations between the collection and other variables should be from the same provider too for the associations to be valid.